test/lib/hub-mock.js

```diff
@@ -0,0 +1,122 @@
+import { CHROME_APP, NOT_FOUND_APP, BASE_URL, BASE_XD_URL } from './constants';
+import { inBrowserstack } from './helpers';
+
+const apps = [];
+
+let hub;
+
+export const windowMapper = {
+	appByWindow(win) {
+		if (win === window.top) {
+			return CHROME_APP;
+		}
+
+		if (apps.length) {
+			const frames = window.frames;
```

Comment:
This seems a bit brittle. 

If we rely on `frame#id` in the `windowByApp` method, why not use it in the reverse? e.g.

```js
const id = win.frameElement.id;
return apps[id];
```

Reply:
That will throw `Uncaught DOMException: Blocked a frame with origin "http://localhost:9000" from accessing a cross-origin frame.` for all frames except the same-origin non-sandboxed.